linalg/hypre.cpp

```diff
@@ -5243,11 +5244,191 @@ void HypreAMS::Init(ParFiniteElementSpace *edge_fespace)
    MakeGradientAndInterpolation(edge_fespace, cycle_type);
 }
 
+void HypreAMS::ResetAMSPrecond()
+{
+   /// This would be closer to an ideal implementation, but the `hypre_AMSData`
+   /// type and associated macros are defined in Hypre's private header ams.h,
+   /// so we can't use their definitions here. I opened an issue:
+   /// https://github.com/hypre-space/hypre/issues/667 to address that.
+   /// This approach is closer to HypreBoomerAMG::ResetAMGPrecond, and "better"
+   /// since it would read the values previously set and use those in the newly
+   /// built AMS solver, indead of having to resort to the defaults.
```

Comment:
I think the current implementation is sufficient, because there are very few AMS parameters that we expose to the user.

I will like to avoid the code duplication though, by having `ResetAMSPrecond` call `MakeSolver`. I'll propose a PR towards this one for that.